compiler/resolution/callDestructors.cpp

```diff
@@ -609,15 +616,22 @@ void ReturnByRef::transformMove(CallExpr* moveExpr)
   if (copyExpr) {
     FnSymbol* rhsFn = copyExpr->isResolved();
 
-    copyExpr->replace(copyExpr->get(1)->remove());
-
-    // But... if we're replacing an init copy, we got to a user
-    // variable, so add an unalias call if there is one
-    if (rhsFn->hasFlag(FLAG_INIT_COPY_FN)) {
-      FnSymbol* unaliasFn = getUnalias(useLhs->type);
-      if (unaliasFn) {
-        callExpr->insertAfter(new CallExpr(unaliasFn, refVar));
-      }
+    // If replacing an init copy, we got to a user variable. Use an unalias
+    // call if possible
+    if (rhsFn->hasFlag(FLAG_INIT_COPY_FN) && unaliasFn != NULL) {
+      // BHARSH: It seems important that there's a temporary to store the
+      // result of the unaliasFn call. Otherwise we'll move into a variable
+      // that has multiplie uses, which seems to cause a variety of problems.
+      //
+      // In particular, I noticed that `changeRetToArgAndClone` would generate
+      // bad AST if I simply did this:
+      //   copyExpr->replace(new CallExpr(unaliasFn, refVar));
```

Comment:
We ought to completely remove changeRetToArgAndClone, but it does special things for initCopy/autoCopy and maybe those special things would need to apply to chpl__unalias too. Nonetheless this workaround is OK with me if it resolves the issue.